### PR TITLE
Fixing state map adding secrets

### PR DESCRIPTION
**Pull request by cwatsonc**
#### Checklist

* [X] Use a separate branch in your local repo (not `master`).
* [X] I've read and agree with the [contribution guidelines](https://www.freeboardgames.org/docs/?path=/story/documentation-contributing--page).

### CODE REVIEW

File: web/src/games/cardtable/game.ts [hunk 1]
```diff
@@ -274,7 +304,11 @@ const rotateTurnToDeal = (G: IG, ctx: Ctx) => {
   G.deck.push(G.deck[0]); //kludge to face the turn
   G.bestCut = null;
   G.cutTie = false;
-  ctx.events.endTurn();
+
+  // eslint-disable-next-line no-console
+  console.log(`ctx.currentPlayer ${ctx.currentPlayer}`);
```
Comment (vdfdev) [inline web/src/games/cardtable/game.ts:309]: Please remove console.log

File: web/src/games/cardtable/game.ts [hunk 2]
```diff
@@ -80,6 +80,36 @@ export interface ICardMove {
   to: ILocation;
 }
 
+const mapForPlayer = (G: IG, ctx: Ctx, id: string) => {
+  // eslint-disable-next-line no-console
+  console.log(`ID: ${id} typeof ID ${typeof id}`);
```
Comment (vdfdev) [inline web/src/games/cardtable/game.ts:85]: Please remove console.log

File: web/src/games/cardtable/DeckContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+  const handleDeal = props.collaborator.handlers.handleDeal;
+  const handleCutDeal = props.collaborator.handlers.handleCutDeal;
+  const handleCutTurn = props.collaborator.handlers.handleCutTurn;
+  const handleRotate = props.collaborator.handlers.handleRotateTurn;
+  const [isOpen, setOpen] = React.useState(false);
+  const [selectedValue, setSelectedValue] = React.useState(19);
+
+  const isClickAllowed = (parentContainer: string): boolean => {
+    if (stage && parentContainer) {
```
Comment (vdfdev) [inline web/src/games/cardtable/DeckContainer.tsx:56]: It is better to use guard statements to reduce nesting: if (!stage || !parentContainer) { return }... https://en.wikipedia.org/wiki/Guard_(computer_science)

File: web/src/games/cardtable/CribContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        }
+        default:
+          return false;
+      }
+    }
+  };
+
+  const handleClick = (idx?: number, name?: string) => {
+    let allowed: boolean = isClickAllowed(name);
+    if (stage && allowed) {
```
Comment (vdfdev) [inline web/src/games/cardtable/CribContainer.tsx:76]: Ditto

File: web/src/games/cardtable/CribContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  const handleCutTurn = props.collaborator.handlers.handleCutTurn;
+  const handlePlay = props.collaborator.handlers.handlePlay;
+  const handleFlip = props.collaborator.handlers.handleFlip;
+  const handleRotate = props.collaborator.handlers.handleRotateTurn;
+  const [isOpen, setOpen] = React.useState(false);
+  const [selectedValue, setSelectedValue] = React.useState(19);
+
+  const isClickAllowed = (parentContainer: string): boolean => {
+    // eslint-disable-next-line no-console
+    console.log(
```
Comment (vdfdev) [inline web/src/games/cardtable/CribContainer.tsx:60]: Ditto

**Review by vdfdev — APPROVED** [review]

File: web/src/games/cardtable/DeckContainer.tsx (hunk 3 above)
Comment (cwatsonc) [reply web/src/games/cardtable/DeckContainer.tsx:56]: originally this was due to a higher abstraction serving multiple implementations of CardContainer.  I always favor single exit points with methods/functions to having multiple return statements.  I believe the downstream logic predicates on these states and additional conditions -- will look for a method to refactor.

File: web/src/games/cardtable/CribContainer.tsx (hunk 4 above)
Comment (cwatsonc) [reply web/src/games/cardtable/CribContainer.tsx:76]: ibid

File: web/src/games/cardtable/CribContainer.tsx (hunk 5 above)
Comment (cwatsonc) [reply web/src/games/cardtable/CribContainer.tsx:60]: ibid

**Pull request merged**